mbed/source/main.cpp

```diff
@@ -0,0 +1,423 @@
+    }
+
+    bool have_periodic_advertising()
+    {
+        return _ble.gap().isFeatureSupported(ble::controller_supported_features_t::LE_EXTENDED_ADVERTISING)
+            && _ble.gap().isFeatureSupported(ble::controller_supported_features_t::LE_PERIODIC_ADVERTISING)
+        ;
+    }
+
+    /** Set up and start advertising */
+    void advertise()
+    {
+        update_state(bt_test_state_t::ADVERTISE);
+        ble_error_t error;
+        if (_is_periodic) {
+            // Perform feature test.
+            if (!have_periodic_advertising()) {
+                printf("Periodic advertising not supported, cannot run test.\r\n");
+                return;
+            }
+
+            // Set advertising parameters. We only do this once, as it allocates memory and we do not call 
+            // destroyAdvertisingSet.
+            if (!_have_adv_handle) {
+                ble::AdvertisingParameters adv_parameters(ble::advertising_type_t::NON_CONNECTABLE_UNDIRECTED);
+                adv_parameters.setUseLegacyPDU(false);
+
+                error = _ble.gap().createAdvertisingSet(&_adv_handle, adv_parameters);
+                if (error) {
+                    print_error(error, "Gap::createAdvertisingSet() failed");
+                    return;
+                }
+
+                error = _ble.gap().setAdvertisingParameters(_adv_handle, adv_parameters);
+                if (error) {
+                    print_error(error, "Gap::setAdvertisingParameters() failed");
+                    return;
+                }
+
+                _have_adv_handle = true;
+            }
+        } else {
+            // Use legacy advertising handle.
+            _adv_handle = ble::LEGACY_ADVERTISING_HANDLE;
+        }
+
+        // Set advertising payload.
+        _adv_data_builder.setFlags();
+        _adv_data_builder.setName(DEVICE_NAME);
+        error = _ble.gap().setAdvertisingPayload(_adv_handle, _adv_data_builder.getAdvertisingData());
+        if (error) {
+            print_error(error, "Gap::setAdvertisingPayload() failed");
+            return;
+        }
+
+        // Start advertising.
+        error = _ble.gap().startAdvertising(_adv_handle, _advertise_time);
+        if (error) {
+            print_error(error, "Gap::startAdvertising() failed");
+            return;
+        }
+    }
+
+    /** Set up and start scanning */
+    void scan()
+    {
+        update_state(bt_test_state_t::SCAN);
+
+        ble::ScanParameters scan_params;
+        scan_params.setOwnAddressType(ble::own_address_type_t::RANDOM);
+
+        ble_error_t error = _ble.gap().setScanParameters(scan_params);
+        if (error) {
+            print_error(error, "Gap::setScanParameters failed");
+            return;
+        }
+
+        error = _ble.gap().startScan(_scan_time);
+        if (error) {
+            print_error(error, "Gap::startScan failed");
+            return;
+        }
+
+        printf("Scanning for %ums\r\n", _scan_time.valueInMs());
```

Comment:
```suggestion
        printf("Scanning for %lums\r\n", _scan_time.valueInMs());
```
or `PRIu64` perhaps?

Reply:
I think `%u` is correct, as `valueInMs` returns `uint32_t`.

Reply:
`PRIu32` is the correct answer :P.